### PR TITLE
fix potential `ArgumentException` during shard rebalancing

### LINKED ISSUE #7365: Akka.Cluster.Sharding: `LeastShardAllocationStrategy` `ArgumentException` during rebalance
**Version Information**
Version of Akka.NET? not sure yet - waiting on reply in Discord
Which Akka.NET Modules? Akka.Cluster.Sharding

**Describe the bug**

```
System.ArgumentException: An element with the same key but a different value already exists. Key: 'akka.tcp://NuBreedSys@prod-nubreed-quote-process-managers-0.prod-nubreed-quote-process-managers-service:9233'
   at System.Collections.Immutable.ImmutableDictionary2.HashBucket.Add(TKey key, TValue value, IEqualityComparer1 keyOnlyComparer, IEqualityComparer1 valueComparer, KeyCollisionBehavior behavior, OperationResult& result)
   at System.Collections.Immutable.ImmutableDictionary2.AddRange(IEnumerable1 items, MutationInput origin, KeyCollisionBehavior collisionBehavior)
   at Akka.Cluster.Sharding.Internal.AbstractLeastShardAllocationStrategy.RegionEntriesFor(IImmutableDictionary2 currentShardAllocations)
   at Akka.Cluster.Sharding.LeastShardAllocationStrategy.Rebalance(IImmutableDictionary2 currentShardAllocations, IImmutableSet1 rebalanceInProgress)
   at Akka.Cluster.Sharding.ShardCoordinator.Active(Object message)
   at Akka.Cluster.Sharding.DDataShardCoordinator.<Activate>b__45_0(Object msg)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Phobos.Actor.PhobosActorCell.ReceiveMessage(Object message)
   at Akka.Actor.ActorCell.Invoke(Envelope envelope)

```

**Expected behavior**

**Actual behavior**
What actually happened and how did it differ from your expectations?

**Screenshots**

**Environment**

Using `state-store-mode = ddata`

**Additional context**

Not sure yet, still working on getting thoughts organized here

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

close #7365

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #7365

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Sharding/ShardAllocationStrategy.cs [hunk 1]
```diff
@@ -169,11 +169,11 @@ public override Task<IImmutableSet<ShardId>> Rebalance(IImmutableDictionary<IAct
                 var sortedRegionEntries = RegionEntriesFor(currentShardAllocations).OrderBy(i => i, ShardSuitabilityOrdering.Instance).ToImmutableList();
                 if (IsAGoodTimeToRebalance(sortedRegionEntries))
                 {
-                    var (_, Shards) = MostSuitableRegion(sortedRegionEntries);
+                    var mostSuitableRegion = MostSuitableRegion(sortedRegionEntries);
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding/ShardAllocationStrategy.cs]: fixed a casing convention here

File: src/contrib/cluster/Akka.Cluster.Sharding/Internal/AbstractLeastShardAllocationStrategy.cs [hunk 2]
```diff
@@ -126,7 +126,13 @@ protected bool IsAGoodTimeToRebalance(IEnumerable<RegionEntry> regionEntries)
 
         protected ImmutableList<RegionEntry> RegionEntriesFor(IImmutableDictionary<IActorRef, IImmutableList<string>> currentShardAllocations)
         {
-            var addressToMember = ClusterState.Members.ToImmutableDictionary(m => m.Address, m => m);
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding/Internal/AbstractLeastShardAllocationStrategy.cs:129]: Left some comments that explain the change

**Review by Aaronontheweb — COMMENTED** [review]
Detailed my changes

**Comment by Aaronontheweb** [comment]
I had to manually kick off AzDo pipelines here, not sure why

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**